### PR TITLE
Cross compile for armv7/armv8, raise raspi variant to armv6

### PR DESCRIPTION
**Pull request by hardcore-coder**
Hi,

I extended the stash cross compilation to include newer ARM platforms (current raspis and NAS'). I used the [naming convention of the official docker images](https://github.com/docker-library/official-images#architectures-other-than-amd64)) but kept the existing binary file names for backwards compatibility.

In addition, I added Dockerfiles for ARM. They are practically all the same but I decided not to put too much effort in it before I got some feedback. What we probably want is taking advantage of the [docker multi arch support](https://docs.docker.com/docker-for-mac/multi-arch/).

Disclaimer: I tested only the arm64v8 version and it's working as expected. The arm32v7 variant compiles but my newer raspis are not available for testing atm.

I don't know why the raspi version was compiled with GOARM=5, as far as I know (and could google) the first raspi supports v6, so I increased this. Did you have a specific reason to choose v5 over v6? If yes, I will revert this change.

I have not updated the `upload-pull-request.sh` script because the raspi version was already missing there.

I guess this pull request will hit this issue: #352 

Fixes: #585

### CODE REVIEW

**Comment by hardcore-coder** [comment]
one more thing.
I stumbled upon the user being `stashappdev` in the [compiler makefile](https://github.com/stashapp/stash/blob/develop/docker/compiler/Makefile#L1) and `stashapp` in the cross compile script. It took me a moment to realize why the cross compilation did not pick up my changes to the compiler container during local development.

File: .travis.yml [hunk 1]
```diff
@@ -44,6 +44,8 @@ deploy:
     - dist/stash-osx
     - dist/stash-win.exe
     - dist/stash-linux
+    - stash-linux-arm64v8
```
Comment (bnkai) [inline .travis.yml:47]: - dist/stash-linux-arm64v8

File: .travis.yml [hunk 2]
```diff
@@ -44,6 +44,8 @@ deploy:
...
+    - stash-linux-arm32v7
```
Comment (bnkai) [inline .travis.yml:48]: `- dist/stash-linux-arm32v7`

File: .travis.yml [hunk 3]
```diff
@@ -61,6 +63,8 @@ deploy:
     - dist/stash-osx
     - dist/stash-win.exe
     - dist/stash-linux
+    - stash-linux-arm64v8
```
Comment (bnkai) [inline .travis.yml:66]: `- dist/stash-linux-arm64v8`

File: .travis.yml [hunk 4]
```diff
@@ -61,6 +63,8 @@ deploy:
...
+    - stash-linux-arm32v7
```
Comment (bnkai) [inline .travis.yml:67]: `- dist/stash-linux-arm32v7`

**Comment by bnkai** [comment]
I don't remember why the version was 5 but i'm guessing for broader compatibility (there are also other arm platforms excpet raspi)? If we take the raspi as a baseline i think armv6 is acceptable.
Also any docker related PR as you mentioned won't complete as the docker image needs to be pushed beforehand.
The `upload-pull-request.sh` is mainly used to help try/debug PRs so leaving as it is isok.

File: docker/production/x86_64/Dockerfile [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
+RUN curl -L -o /stash $(curl -s https://api.github.com/repos/stashapp/stash/releases/latest | awk '/browser_download_url/ && /$STASH_BINARY/' | sed -e 's/.*: "\(.*\)"/\1/') && \
     chmod +x /stash && \
-    curl --http1.1 -o /ffmpeg.tar.xz https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.tar.xz && \
+    curl --http1.1 -o /ffmpeg.tar.xz https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-${FFMPEG_ARCH}-static.tar.xz && \
     tar xf /ffmpeg.tar.xz && \
     rm ffmpeg.tar.xz && \
     mv /ffmpeg*/ /ffmpeg/
 
-FROM ubuntu:18.04 as app
-RUN apt-get update && apt-get -y install ca-certificates
```
Comment (bnkai) [inline docker/production/x86_64/Dockerfile:18]: any reason to change that?
copying the certs as you do later should work but isn't it better to have the package installed?

File: .travis.yml (hunk 4 above)
Comment (bnkai) [inline .travis.yml:67]: - dist/stash-linux-arm32v7

File: docker/production/x86_64/Dockerfile (hunk 5 above)
Comment (bnkai) [reply docker/production/x86_64/Dockerfile:18]: Fair enough. Seems ok then.

**Comment by bnkai** [comment]
> Would it make sense to move the changes in the compiler Dockerfile to a separate PR? When that is merged travis should be able to test this PR, no?

I think that would be better. It will be easier to troubleshoot also if we have the compiler docker image available.

**Comment by WithoutPants** [comment]
I've opened #737 to just include the cross-compilation changes. Thanks for the contribution. I'll close this and you can open a new one for the dockerfile changes.

**Pull request closed without merging**